chrome-launcher/chrome-launcher.ts

```diff
@@ -45,24 +48,79 @@ export interface ModuleOverrides {
   spawn?: typeof childProcess.spawn;
 }
 
-export async function launch(opts: Options = {}): Promise<LaunchedChrome> {
-  opts.handleSIGINT = defaults(opts.handleSIGINT, true);
+/**
+ * LaunchManager handles launch retries and SIGINT binding
+ */
+export class LaunchManager {
+  private instance?: Launcher;
+  private opts: Options;
+  private isSigintBound: boolean;
+  private maxLaunchAttempts: number;
+  private attempts: number;
+
+  constructor() {
+    this.isSigintBound = false;
+    this.maxLaunchAttempts = 3;
+    this.attempts = 1;
+  }
+
+  setOptions(opts: Options = {}) {
+    this.opts = opts;
+  }
+
+  async launchInstance(): Promise<LaunchedChrome> {
+    if (this.instance) {
+      Promise.reject(new Error('Already a launcher instance, can\'t create a second.'));
```

Comment:
need to return this?

unrelated I wonder if ts async is fancy enough to wrap the whole thing in a `promise.resolve().then()` so we could just throw it